app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -213,8 +213,8 @@ public void onCloseRequest(GeckoSession geckoSession) {
             };
         }
 
-        private GeckoSession.ProgressListener createProgressListener() {
-            return new GeckoSession.ProgressListener() {
+        private GeckoSession.ProgressDelegate createProgressDelegate() {
```

Comment:
\[PMD\] INFO: The method 'createProgressDelegate' has a Cyclomatic Complexity of 7\.